File: AspNetCore/AspNetCore/Models/Entity.cs
namespace AspNetCore.Models
{
    public class Entity {
        public string id { get; set; }
        public int document_type { get; set; }
        public string[] string_array { get; set; }
        public int[] int_array { get; set; }
        public SubEntity[] child_objects { get; set; }
    }

    public class SubEntity {
        public string id { get; set; }
        public string name { get; set; }
        public int number { get; set; }
    }
}

File: frameworks/nestjs-fastify/src/entities/Entity.ts
import { SubEntity } from './SubEntity';
import { ApiProperty } from '@nestjs/swagger';

export class Entity {
  @ApiProperty()
  id: string;
  @ApiProperty()
  type: number;
  @ApiProperty({ type: [String] })
  string_array: string[];
  @ApiProperty({ type: [Number] })
  int_array: number[];
  @ApiProperty({ type: [SubEntity] })
  child_objects: SubEntity[];
}
